main/kotlin/com/example/people_managment_system/controller/SystemController.kt

```diff
@@ -0,0 +1,52 @@
+package com.example.people_managment_system.controller
+
+import com.example.people_managment_system.dtos.EmployeeDto
+import com.example.people_managment_system.service.SystemService
+import org.springframework.dao.DataIntegrityViolationException
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.*
+
+@RestController
+@RequestMapping("api/system")
+
+class SystemController(val service: SystemService)
+    {
+    @ExceptionHandler(NoSuchElementException::class)
+    fun handleNotFound(e: NoSuchElementException): ResponseEntity<String> =
+        ResponseEntity(e.message, HttpStatus.NOT_FOUND)
+
+    @ExceptionHandler(DataIntegrityViolationException::class)

```

Comment:
Exception handling is always a must and it's good that you added it in here.

What I would recommend is creating a class called `ExceptionHandler` and give it the `@RestControllerAdvice` annotation. 

Think of this class as the source of truth for handling exceptions in your service.

You can create methods to handle exceptions and tell the method what type of a `ResponseEntity` you'd like to return back to the user.

Check this link out for more information: https://codersee.com/exception-handling-with-restcontrolleradvice-and-exceptionhandler/